llvm/lib/SYCLLowerIR/GlobalOffset.cpp

```diff
@@ -59,11 +59,21 @@ ModulePass *llvm::createGlobalOffsetPassLegacy() {
   return new GlobalOffsetLegacy();
 }
 
+void getLoads(Instruction *P, SmallVectorImpl<Instruction *> &Traversed,
+              SmallVectorImpl<LoadInst *> &Loads) {
+  Traversed.push_back(P);
+  auto *L = dyn_cast<LoadInst>(P);
```

Comment:
I quite like the idiom of folding your variable into the `if`condition, especially with `dyn_cast`s, something along the lines of:
```cpp
if (auto *L = dyn_cast<LoadInst>(P))
```